### PR TITLE
Forest Fire Control Panel

### PR DESCRIPTION
**Pull request by EmilyHu1224**
Implemented the control panel for the forest fire model:
image

Container:
* Model Header (title & description)
* Simulation Panel
* Control Panel
    * Basic Control Fields (grid size & neighborhood)
    * State Definitions (color & proportion per agent state)
    * Evolution Rules (chance of lightning & reviving)

### CODE REVIEW

File: src/components/ForestFire/ForestFireStateDefinitions.tsx [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+  const { className, stateDefinitions, setStateDefinitions } = props;
+  const classes = useStyles();
+  const onDefinitionChange = (attribute: attr, index: number, e: React.ChangeEvent<HTMLInputElement>) => {
+    setStateDefinitions(definitions => {
+      switch (attribute) {
+        case attr.COLOR:
+          definitions[index].color = e.target.value;
+        case attr.PROPORTION:
+          definitions[index].proportion = Number(e.target.value);
+      };
```
Comment (EmilyHu1224) [inline src/components/ForestFire/ForestFireStateDefinitions.tsx]: There are 3 agent states with  2 editable attributes (color and proportion) per state, stored as an array of objects.
Each edit action needs to specify the target **agent state** and the **attribute**. The **agent state** can be identified using the array index, and I defined `enum attr` to control the *attribute*.
Would like to know if there's a more elegant way to do this - e.g., accessing object attributes using strings?

Comment (EmilyHu1224) [reply src/components/ForestFire/ForestFireStateDefinitions.tsx]: UGH I forgot to add `break` lol

Comment (EmilyHu1224) [reply src/components/ForestFire/ForestFireStateDefinitions.tsx]: also this doesn't really work :(((

File: src/components/ForestFire/ForestFireEvolutionRules.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+            min: 0,
+            max: 100,
+          }}
+          value={pReviving}
+          onChange={changePReviving}
+          endAdornment={<span>%</span>}
+          className={classes.chanceInput}
+        />
+      </FormGroup>
+      <Table className={classes.table}>
```
Comment (cbwestaway) [inline src/components/ForestFire/ForestFireEvolutionRules.tsx]: Move table to separate component which generates it programatically

File: src/components/ForestFire/ForestFireStateDefinitions.tsx (hunk 1 above)
Comment (cbwestaway) [reply src/components/ForestFire/ForestFireStateDefinitions.tsx]: I would break this logic into two separate onChange functions.

**Review by cbwestaway — CHANGES_REQUESTED** [review]

**Review by cbwestaway — APPROVED** [review]

**Pull request merged**